internal/types/regex.go

```diff
@@ -21,17 +21,39 @@ import (
 )
 
 var (
+	// ErrOptionNotImplemented indicates unimplemented regex option.
 	ErrOptionNotImplemented = fmt.Errorf("regex: option not implemented")
-	ErrMissingParen         = fmt.Errorf("Regular expression is invalid: missing )")
-	ErrMissingBracket       = fmt.Errorf("Regular expression is invalid: missing terminating ] for character class")
-	ErrInvalidEscape        = fmt.Errorf("Regular expression is invalid: PCRE does not support \\L, \\l, \\N{name}, \\U, or \\u")
-	ErrMissingTerminator    = fmt.Errorf("Regular expression is invalid: syntax error in subpattern name (missing terminator)")
+
+	// ErrMissingParen indicates missing parentheses in regex expression.
+	ErrMissingParen = fmt.Errorf("Regular expression is invalid: missing )")
+
+	// ErrMissingBracket indicates missing terminating ] for character class.
+	ErrMissingBracket = fmt.Errorf("Regular expression is invalid: missing terminating ] for character class")
+
+	// ErrInvalidEscape indicates invalid escape errors.
+	ErrInvalidEscape = fmt.Errorf("Regular expression is invalid: PCRE does not support \\L, \\l, \\N{name}, \\U, or \\u")
+
+	// ErrMissingTerminator indicates syntax error in subpattern name (missing terminator)..
+	ErrMissingTerminator = fmt.Errorf("Regular expression is invalid: syntax error in subpattern name (missing terminator)")
+
+	// ErrUnmatchedParentheses indicates unmatched parentheses.
 	ErrUnmatchedParentheses = fmt.Errorf("Regular expression is invalid: unmatched parentheses")
-	ErrTrailingBackslash    = fmt.Errorf("Regular expression is invalid: \\ at end of pattern")
-	ErrNothingToRepeat      = fmt.Errorf("Regular expression is invalid: nothing to repeat")
-	ErrInvalidClassRange    = fmt.Errorf("Regular expression is invalid: range out of order in character class")
-	ErrUnsupportedPerlOp    = fmt.Errorf("Regular expression is invalid: unrecognized character after (? or (?-")
-	ErrInvalidRepeatSize    = fmt.Errorf("Regular expression is invalid: regular expression is too large")
+
+	// ErrTrailingBackslash indicates \\ at end of the pattern.
+	ErrTrailingBackslash = fmt.Errorf("Regular expression is invalid: \\ at end of pattern")
+
+	// ErrNothingToRepeat indicates invalid regex: nothing to repeat.
+	ErrNothingToRepeat = fmt.Errorf("Regular expression is invalid: nothing to repeat")
+
+	// ErrInvalidClassRange indicates that range out of order in character class.
+	ErrInvalidClassRange = fmt.Errorf("Regular expression is invalid: range out of order in character class")
+
+	//nolint:godot // false positive
+	// ErrUnsupportedPerlOp indicates for unrecognized character after (? or (?-.
```

Comment:
This could be rephrased:
```suggestion
	// ErrUnsupportedPerlOp indicates unrecognized character after the grouping sequence start.
```